flak-spi/src/main/java/flak/spi/AbstractApp.java

```diff
@@ -219,4 +221,14 @@ public Request getRequest() {
   public Response getResponse() {
     return getRequest().getResponse();
   }
+
+  public void addBeforeAllHook(BeforeHook hook) {
```

Comment:
Shouldn't this be a public method in App? It would deserve an @Override

Reply:
`BeforeHook` is defined in `flak-spi` and `App` in `flak-api`, and to avoid cyclic dependencies, `flak-spi` seams to depend to `flak-api` but not the other way (i.e. I can't do that without moving `BeforeHook` or using another dedicated interface).

Reply:
Ok, forgot that BeforeHook was not public API